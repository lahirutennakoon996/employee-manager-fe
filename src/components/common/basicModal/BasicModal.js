export default function BasicModal({title, body, cancelText, confirmText, confirmAction}) {
  return (
    <div className="modal fade" id="basicModal" tabIndex="-1">
      <div className="modal-dialog">
        <div className="modal-content">
          <div className="modal-header">
            <h1 className="modal-title fs-5" id="basicModalLabel">{title}</h1>
            <button type="button" className="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
          </div>
          <div className="modal-body">
            {body}
          </div>
          <div className="modal-footer">
            <button type="button" className="btn btn-secondary" data-bs-dismiss="modal">{cancelText}</button>
            <button type="button" className="btn btn-danger" data-bs-dismiss="modal" onClick={confirmAction}>
              {confirmText}
            </button>
          </div>
        </div>
      </div>
    </div>
  )
}