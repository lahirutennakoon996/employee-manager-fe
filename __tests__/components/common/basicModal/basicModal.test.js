import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

import BasicModal from "@/components/common/basicModal/BasicModal";
import mockStore from "../../../../__mocks__/react-redux.mock";

describe('Basic modal', () => {
  const modalData = {
    title: "Are you sure?",
    body: "This will delete the employee from the system.",
    cancelText: "Cancel",
    confirmText: "Yes",
  };

  const initialState = {
    employees: {
      allEmployees: [],
      singleEmployee: null,
    },
  };

  const store = mockStore(initialState);
  let component;

  beforeEach(() => {
    const componentRendered = render(<Provider store={store}>
      <BasicModal
        title={modalData.title}
        body={modalData.body}
        cancelText={modalData.cancelText}
        confirmText={modalData.confirmText}
        confirmAction={jest.fn()}
      />
    </Provider>);
    component = componentRendered.container;
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('render the title', () => {
    const heading = screen.getByRole('heading', {
      name: modalData.title,
    });
    expect(heading).toBeInTheDocument();
  })

  it('render the close button', () => {
    const closeBtn = screen.getByRole('button', {
      name: /close/i,
    });
    expect(closeBtn).toBeInTheDocument();
    expect(closeBtn).not.toBeDisabled();
  })

  it('render the confirm button', () => {
    const confirmBtn = screen.getByRole('button', {
      name: modalData.confirmText,
    });
    expect(confirmBtn).toBeInTheDocument();
    expect(confirmBtn).not.toBeDisabled();
  })

  it('render the cancel button', () => {
    const cancelBtn = screen.getByRole('button', {
      name: modalData.cancelText,
    });
    expect(cancelBtn).toBeInTheDocument();
    expect(cancelBtn).not.toBeDisabled();
  })
})