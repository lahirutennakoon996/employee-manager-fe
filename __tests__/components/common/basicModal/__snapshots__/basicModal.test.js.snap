// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic modal should match snapshot 1`] = `
<div>
  <div
    class="modal fade"
    id="basicModal"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h1
            class="modal-title fs-5"
            id="basicModalLabel"
          >
            Are you sure?
          </h1>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          This will delete the employee from the system.
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Cancel
          </button>
          <button
            class="btn btn-danger"
            data-bs-dismiss="modal"
            type="button"
          >
            Yes
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
